plugins/cairo-lang-macro-stable/src/ffi.rs

```diff
@@ -0,0 +1,44 @@
+/// This struct encapsulates a stable ABI representation of struct.
+///
+/// Please be aware that the memory management of values passing the FFI-barrier is tricky.
+#[repr(C)]
+#[derive(Debug)]
+#[doc(hidden)]
+pub struct StableSlice<T> {
+    ptr: *mut T,
+    len: usize,
+}
+
+impl<T> Copy for StableSlice<T> {}
+
+impl<T> Clone for StableSlice<T> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<T> StableSlice<T> {
+    /// Create a new `StableSlice` from a Vector.
+    /// Note that the vector will not be deallocated automatically.
+    /// Please make sure to use `into_owned` afterward, to free the memory.
+    pub fn new(mut x: Vec<T>) -> Self {
+        x.shrink_to_fit();
+        assert_eq!(x.len(), x.capacity());
```

Comment:
Is this actually a guaranteed property of `shrink_to_fit`? Docs do not say that, If it is not we probably have to store capacity as well in the `StableSlice`? 🤔 Not sure though